cicd/builds/triggering-builds-build-hooks.adoc

```diff
@@ -35,3 +35,5 @@ include::modules/builds-build-hooks.adoc[leveloffset=+1]
 include::modules/builds-configuring-post-commit-build-hooks.adoc[leveloffset=+2]
 
 include::modules/builds-using-cli-post-commit-build-hooks.adoc[leveloffset=+2]
+
+include::modules/unauthenticated-users-cluster-role-binding.adoc[leveloffset=+1]
```

Comment:
It would be good if we would not use the general module here, but a dedicated one. We could add a note / a highlighted box that this change / step was introduced with 4.16 and that it is a measure that improves the security.

**We would need something like:**

[Box: this is new with 4.16]
[Box: [IMPORTANT] Always verify compliance with your organization's security standards when modifying unauthenticated access.]

By default there is no access for systems that don't use OpenShift's means of Authentication, like Service Account Tokens. Therefore there is a need to open up the namespaces, that need the ability to be triggered by unauthenticated WebHooks like from GitHub with a RoleBinding like this:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  name: webhook-access-unauthenticated
  namespace: $YOUR_NAMESPACE
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: "system:webhook"
subjects:
  - apiGroup: rbac.authorization.k8s.io
    kind: Group
    name: "system:unauthenticated"
```

If opening up individual namespaces with a RoleBinding is not feasible, all namespaces can be opened up by default with a ClusterRoleBinding. This is not recommended as it creates an unnecessarily wide attack surface. The ClusterRoleBinding looks like:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
 annotations:
   rbac.authorization.kubernetes.io/autoupdate: "true"
 name: webhook-access-unauthenticated
roleRef:
 apiGroup: rbac.authorization.k8s.io
 kind: ClusterRole
 name: "system:webhook"
subjects:
 - apiGroup: rbac.authorization.k8s.io
   kind: Group
   name: "system:unauthenticated"
```.